src/Arduino_NotecardConnectionHandler.cpp

```diff
@@ -0,0 +1,979 @@
+  _notecard{},
+  _device_id{},
+  _notecard_uid{},
+  _notehub_url(notehub_url_),
+  _project_uid(project_uid_)
+{ }
+
+NotecardConnectionHandler::NotecardConnectionHandler(
+  const String & project_uid_,
+  HardwareSerial & serial_,
+  uint32_t speed_,
+  bool en_hw_int_,
+  bool keep_alive_,
+  const String & notehub_url_
+) :
+  ConnectionHandler{keep_alive_, NetworkAdapter::NOTECARD},
+  _serial(&serial_),
+  _wire(nullptr),
+  _inbound_buffer(nullptr),
+  _conn_start_ms(0),
+  _i2c_address(0),
+  _i2c_max(0),
+  _uart_speed(speed_),
+  _inbound_buffer_index(0),
+  _inbound_buffer_size(0),
+  _en_hw_int(en_hw_int_),
+  _topic_type{TopicType::Invalid},
+  _notecard{},
+  _device_id{},
+  _notecard_uid{},
+  _notehub_url(notehub_url_),
+  _project_uid(project_uid_)
+{ }
+
+/******************************************************************************
+   PUBLIC MEMBER FUNCTIONS
+ ******************************************************************************/
+
+const String & NotecardConnectionHandler::syncArduinoDeviceId (const String & device_id_)
+{
+  // Validate the connection state is not in an initialization state
+  if (check() == NetworkConnectionState::INIT)
+  {
+    Debug.print(DBG_ERROR, F("Failed to sync Arduino Device ID. Connection has not been initialized."));
+    return device_id_;
+  }
+
+  // Prefer the incoming Device ID. If the value is known to be invalid, then
+  // fetch the Device ID from the Notecard Connection Handler (cloud cache)
+  if ((device_id_ == "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx" || device_id_.length() == 0)
+  && (updateUidCache() && _device_id.length() > 0)) {
+    Debug.print(DBG_VERBOSE, F("Resolved Arduino Device ID with cached ID."));
+  } else {
+    _device_id = device_id_;
+    Debug.print(DBG_VERBOSE, F("Resolved Arduino Device ID with provided ID."));
+
+    // Report the Device ID to the Cloud
+    Debug.print(DBG_VERBOSE, F("Updating cache..."));
+    if (J *req = _notecard.newRequest("hub.set"))
+    {
+      if (_device_id.length() > 0) {
+        JAddStringToObject(req, "sn", _device_id.c_str());
+      } else {
+        JAddStringToObject(req, "sn", "-");
+      }
+      if (J *rsp = _notecard.requestAndResponse(req)) {
+        // Check the response for errors
+        if (NoteResponseError(rsp)) {
+          const char *err = JGetString(rsp, "err");
+          Debug.print(DBG_ERROR, F("%s"), err);
+        } else {
+          Debug.print(DBG_VERBOSE, F("Cache updated successfully."));
+        }
+        JDelete(rsp);
+      } else {
+        Debug.print(DBG_ERROR, F("Failed to receive response from Notecard."));
+      }
+    }
+
+    if (_keep_alive && initiateNotehubSync()) {
+      Debug.print(DBG_ERROR, F("Failed to sync Arduino Device ID."));
+    }
+  }
+
+  Debug.print(DBG_DEBUG, F("Synchronized Arduino Device ID: %s"), _device_id.c_str());
+
+  return _device_id;
+}
+
+int NotecardConnectionHandler::syncSecretDeviceKey (const String & secret_device_key_)
```

Comment:
See comment about `device_id`

Reply:
We have removed the concept of device id and secret management from the host device, so this code has been removed completely.